File: 1/lab21.go
package main

import (
	"fmt"
	"strings"
)

func isCyclicShift(S string, T string) bool {

	// Если длины разные, циклический сдвиг невозможен
	if len(S) != len(T) {
		return false
	}

	// Создаём строку из двух S подряд
	doubled := S + S

	// Проверяем встречается ли T внутри SS, если найдена => цикл сдвиг
	if strings.Contains(doubled, T) {
		return true
	}

	return false
}

func main() {

	var S, T string

	fmt.Print("Введите строку S: ")
	fmt.Scan(&S)

	fmt.Print("Введите строку T: ")
	fmt.Scan(&T)

	// Проверка на циклический сдвиг
	if isCyclicShift(S, T) {
		fmt.Println("Yes")
	} else {
		fmt.Println("No")
	}
}

File: 1/lab21.swift
import Foundation

func isCyclicShift(_ S: String, _ T: String) -> Bool {

    // Если длины разные, циклический сдвиг невозможен
    if S.count != T.count {
        return false
    }

    // Создаём строку из двух S подряд
    let doubled = S + S

    // Проверяем встречается ли T внутри SS, если найдена => цикл сдвиг
    if doubled.contains(T) {
        return true
    }

    return false
}

print("Введите строку S: ", terminator: "")
let S = readLine() ?? ""

print("Введите строку T: ", terminator: "")
let T = readLine() ?? ""

// Проверка на циклический сдвиг
if isCyclicShift(S, T) {
    print("Yes")
} else {
    print("No")
}